xc/common/utils/prjxray_import_tile_capacity.py

```diff
@@ -30,70 +47,28 @@ def main():
     db = prjxray.db.Database(args.db_root, args.part)
     tile_type = db.get_tile_type(args.tile_type)
 
-    input_wires = set()
-    output_wires = set()
+    sites = {}
 
-    sites = []
+    pb_types = args.pb_types.split(',')
+
+    for pb_type in pb_types:
+        sites[pb_type] = []
 
     for site in tile_type.get_sites():
-        if site.type != args.pb_type:
+        if site.type not in pb_types:
             continue
 
         site_type = db.get_site_type(site.type)
-        sites.append(site)
-
-        for site_pin in site.site_pins:
-            if site_type.get_site_pin(
-                    site_pin.name).direction == SitePinDirection.IN:
-                input_wires.add(site_pin.wire)
-            elif site_type.get_site_pin(
-                    site_pin.name).direction == SitePinDirection.OUT:
-                output_wires.add(site_pin.wire)
-            else:
-                assert False, site_pin
+        input_wires, output_wires = get_wires(site, site_type)
 
-    sites.sort(key=lambda site: (site.x, site.y))
+        sites[site.type].append((site, input_wires, output_wires))
 
-    tile_xml = start_pb_type(
+    tile_xml = start_tile(
         args.tile_type,
         pin_assignments,
-        input_wires,
-        output_wires,
-        root_pb_type=True,
-        root_tag='tile',
+        sites=sites,
     )
 
-    tile_xml.attrib['capacity'] = str(len(sites))
-    tile_xml.attrib['capacity_type'] = "explicit"
-
-    equivalent_sites_xml = ET.Element('equivalent_sites')
-
-    site_xml = ET.Element(
-        'site', {
-            'pb_type': add_vpr_tile_prefix(site.type),
-            'pin_mapping': 'custom'
-        }
-    )
-    for site_idx, site in enumerate(sites):
-        if site.type != args.pb_type:
-            continue
-
-        for site_pin in site.site_pins:
```

Comment:
This mapping is still needed, where did it go?

Reply:
It is a leftover actually, I tried to move that in the utiliy lib, to extract IO wires. It is in the `get_wires` function.

Reply:
Get wires is the loop here https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1386/files#diff-3c687ed1bdcb39751e14fcd2591aaaadL43-L53, I'm talking about the loop here: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1386/files#diff-3c687ed1bdcb39751e14fcd2591aaaadL81-L93

Reply:
Ah right, I misunderstood. You mean the physical to logical pin mappings. It ended up in the `start_tile` function in https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1386/files#diff-858a400e24b804d76914f3604da75835. I still need to get to better generalize that utility library.